dashboard/app/models/concerns/pd/form.rb

```diff
@@ -0,0 +1,74 @@
+module Pd::Form
+  extend ActiveSupport::Concern
+
+  included do
+    validates_presence_of :form_data
+    validate :validate_required_fields
+    validate :validate_options
+  end
+
+  module ClassMethods
+    def options
+      # should be overridden by including model
+      {}
+    end
+
+    def required_fields
+      # should be overridden by including model
+      []
+    end
+  end
+
+  def add_key_error(key)
+    key = key.to_s.camelize(:lower)
+    errors.add(:form_data, :invalid, message: key)
+  end
+
+  def validate_required_fields
+    hash = form_data_hash.transform_keys {|key| key.underscore.to_sym}
```

Comment:
It's much cleaner if they're underscore case; we want to be comparing these against the results of the `options` and `required_fields`  methods as defined in [the including models](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/facilitator_program_registration.rb); because the data coming up from the server is camelCase, if we didn't undescorize here we'd have to declare those values using `:camelCaseSymbols`